cli/samples_checker.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+import argparse
+import gzip
+import json
+import logging
+import os
+from collections import defaultdict
+
+import yaml
+
+
+def open_gzip_if_required(input_file):
+    if input_file.endswith('.gz'):
+        return gzip.open(input_file, 'rt')
+    else:
+        return open(input_file, 'r')
+
+
+def get_samples_from_vcf(vcf_file):
+    """
+    Get the list of samples present in a single VCF file
+    """
+    with open_gzip_if_required(vcf_file) as vcf_in:
+        for line in vcf_in:
+            if line.startswith('#CHROM'):
+                sp_line = line.strip().split('\t')
+                if len(sp_line) > 9:
+                    return sp_line[9:]
+                else:
+                    logging.warning(f"No Sample names found in file {vcf_file}")
+                    return []
+
+
+def compare_names_in_files_and_samples(sample_name_in_analysis, sample_name_per_file):
```

Comment:
Is there any disadvantage to having all relevant collections of sample names (so I guess `sample_name_in_analysis` and each value in `sample_name_per_file`) be sets rather than lists from the beginning?  I think it would save some code repetition and maybe be faster too.

Reply:
I changed it anyway. Eventually it should replace the one in eva-submission.